### ISSUE TITLE
Duplicate time series doesn't report error in v1.55.0

### ISSUE THREAD

**Issue by arramos84** [VictoriaMetrics/VictoriaMetrics#1141]
Starting in VictoriaMetrics version `v1.55.0` it looks like duplicate timeseries aren't flagged as errors anymore. I have a test which simply checks rules with:

```
kube_pod_container_info * on(pod) up
```

which should return an error. Up to `v1.54.1` I used to see:

```
"error when executing query=\"kube_pod_container_info * on(pod) up\" for (time=1616428017491, step=120000): cannot evaluate \"kube_pod_container_info * on (pod) up\": duplicate time series on the left side of * on (pod): {container=\"grafana\", endpoint=\"http\", instance=\"10.8.90.8:8080\", job=\"kube-state-metrics\", namespace=\"tsdb\", pod=\"grafana-0\", service=\"kube-prometheus-stack-kube-state-metrics\"} and {container=\"grafana-sc-dashboard\", endpoint=\"http\", image=\"kiwigrid/k8s-sidecar:1.1.0\", instance=\"10.8.90.8:8080\", job=\"kube-state-metrics\", namespace=\"tsdb\", pod=\"grafana-0\", service=\"kube-prometheus-stack-kube-state-metrics\"}"
```

Looks like there was a change in the `v1.55.0` release notes saying:

"BUGFIX: reduce the probability of duplicate time series errors when querying Kubernetes metrics."

Guessing this has something to do with it. Built off of master today as well and the query did not fail.

**Comment by valyala** [comment]
@arramos84 , could you share a graph for the query `kube_pod_container_info * on(pod) up`?

**Comment by arramos84** [comment]
I am actually seeing conflicting behavior in grafana. So I'll explain my workflow. I basically have this "bad rule" that looks like this:

```
 cat rules/bad.rules.json | python -m json.tool
{
    "groups": [
        {
            "name": "bad.rules",
            "rules": [
                {
                    "record": "bad_rule",
                    "expr": "kube_pod_container_info * on(pod) up",
                    "labels": {
                        "namespace": "bad_namespace",
                        "job": "bad_job"
                    }
                }
            ]
        }
    ]
}
```

I run it in `vmalert` against victoria-metrics. Keeping the `vmalert` version constant, I change the victoria-metrics to a version that is `v1.55.0` and up and I can never get `vmalert` to throw an error when evaluating this rule. The error would show up in the vmalert logs using previous victoria-metrics versions.

<img width="1698" alt="Screen Shot 2021-03-23 at 1 45 31 PM" src="https://user-images.githubusercontent.com/2239830/112215849-15392800-8bde-11eb-9c66-4e7222ae0aa4.png">



It looks like if I enter the query directly into grafana, it throws the error:


<img width="1706" alt="Screen Shot 2021-03-23 at 1 45 17 PM" src="https://user-images.githubusercontent.com/2239830/112215859-19654580-8bde-11eb-9d71-895419ea8016.png">


So it looks like when used as a recording rule in vmalert, the query passes and writes data.

**Comment by valyala** [comment]
@arramos84 , could you build VictoriaMetrics from the commit 0b2a66db30fc00398144cb374cb23566cdedca35 and verify whether it fixes the issue? See [build instructions](https://victoriametrics.github.io/#how-to-build-from-sources). If you use cluster version of VictoriaMetrics, then build it from the commit b473c21915d27bbf1b64d485ab0c757fc76f494d according to [these docs](https://victoriametrics.github.io/Cluster-VictoriaMetrics.html#building-from-sources).

**Comment by arramos84** [comment]
@valyala looks like it is working as expected now. Thanks!

**Comment by valyala** [comment]
The bugfix has been included in [v1.57.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
